modules/svg/image_loader_svg.cpp

```diff
@@ -81,13 +81,23 @@ void ImageLoaderSVG::create_image_from_string(Ref<Image> p_image, String p_strin
 
 	tvg::Result result = picture->load((const char *)bytes.ptr(), bytes.size(), "svg", true);
 	if (result != tvg::Result::Success) {
-		return;
+		return ERR_INVALID_DATA;
 	}
 	float fw, fh;
 	picture->size(&fw, &fh);
 
-	uint32_t width = MIN(round(fw * p_scale), 16 * 1024);
-	uint32_t height = MIN(round(fh * p_scale), 16 * 1024);
+	uint32_t width = round(fw * p_scale);
+	uint32_t height = round(fh * p_scale);
+
+	const uint32_t max_dimension = 16384;
+	if (width > max_dimension || height > max_dimension) {
+		WARN_PRINT(vformat(
+				String::utf8("ImageLoaderSVG: Canvas dimensions of %d×%d (with scale %.2f) and exceed the max supported dimensions of %d×%d. The canvas will be cropped."),
```

Comment:
Is it cropped or scaled down (probably scaled)?

Reply:
Done.